src/GitVersion.Core/VersionCalculation/TrunkBased/Trunk/CommitOnTrunkBranchedBase.cs

```diff
@@ -6,18 +6,27 @@ namespace GitVersion.VersionCalculation.TrunkBased.Trunk;
 internal abstract class CommitOnTrunkBranchedBase : ITrunkBasedIncrementer
 {
     public virtual bool MatchPrecondition(TrunkBasedIteration iteration, TrunkBasedCommit commit, TrunkBasedContext context)
-        => commit.Configuration.IsMainBranch && commit.BranchName != iteration.BranchName && commit.Successor is null;
+        => commit.GetEffectiveConfiguration(context.Configuration).IsMainBranch
+            && commit.BranchName != iteration.BranchName && commit.Successor is null;
```

Comment:
When statements are broken up into several lines like this, I prefer that every condition is put on its own line.

```suggestion
            && commit.BranchName != iteration.BranchName
            && commit.Successor is null;
```